### PR TITLE
Second try to enable rust async io and io uring

### PR DESCRIPTION
**Pull request by jiangliu**
Try again to enable rust async io and io uring by using tokio-uring.

### CODE REVIEW

**Comment by jiangliu** [comment]
related to： https://github.com/dragonflyoss/image-service/issues/299

**Comment by jiangliu** [comment]
related to：https://github.com/dragonflyoss/image-service/issues/507

File: Cargo.toml [hunk 1]
```diff
@@ -47,10 +46,12 @@ vm-memory = { version = "0.7", features = ["backend-mmap", "backend-bitmap"] }
 
 [features]
 default = ["fusedev"]
-#async-io = ["async-trait", "futures", "iou", "ringbahn", "caps]
+#async-io = ["async-trait", "futures", "caps]
```
Comment (wllenyj) [inline Cargo.toml:49]: remove ?

Comment (jiangliu) [reply Cargo.toml]: Fixed

File: src/transport/file_volatile_slice.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ pub enum Error {
     /// Taking a slice at `base` with `offset` would overflow `usize`.
     Overflow { base: usize, offset: usize },
     /// The error of VolatileSlice.
-    VolatileSliceError(VError),
+    VolatileSlice(VError),
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply src/transport/file_volatile_slice.rs:32]: It fixes a clippy warning

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiangliu) [reply Cargo.toml]: It's a mistake and have been recovered in later patches. I will try to refine my patch:)

File: src/transport/fusedev/mod.rs [hunk 3]
```diff
@@ -354,15 +302,10 @@ impl<'a, S: BitmapSlice> io::Write for Writer<'a, S> {
             if bufs.is_empty() {
                 return Ok(0);
             }
-            writev(self.fd, bufs)
-                .map(|x| {
-                    self.account_written(x);
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply src/transport/fusedev/mod.rs]: It's a mistake and have been recovered in next patch.
I have refined the patchset to fix this confusion change.

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiangliu) [reply Cargo.toml]: dropped by later patches

File: src/transport/file_traits.rs [hunk 4]
```diff
@@ -446,6 +447,607 @@ macro_rules! volatile_impl {
...
+        ) -> (Result<usize>, FileVolatileBuf) {
+            self.read_at(buf, offset).await
+        }
+
+        async fn async_read_vectored_at_volatile(
+            &self,
+            mut bufs: Vec<FileVolatileBuf>,
+            mut offset: u64,
+        ) -> (Result<usize>, Vec<FileVolatileBuf>) {
+            if bufs.len() == 0 {
```
Comment (liubogithub) [inline src/transport/file_traits.rs:526]: Would it be simpler if tokio_uring::File provides read_vectored_at?

**Review by liubogithub — APPROVED** [review]

**Pull request merged**